shared/lib/wp/support.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import qs from 'qs';
+
+export default function wpcomSupport( wpcom ) {
+	let supportUser = '';
+	let supportToken = '';
+
+	/**
+	 * Add the supportUser and supportToken to the query.
+	 * @param {Object}  params The original request params object
+	 * @return {Object}        The new query object with support data injected
+	 */
+	const addSupportData = function( params ) {
+		// Unwind the query string
+		let query = qs.parse( params.query );
+
+		// Inject the credentials
+		query.support_user = supportUser;
+		query._support_token = supportToken
+
+		return Object.assign( {}, params, {
+			query: qs.stringify( query )
+		} );
+	};
+
+	const request = wpcom.request.bind( wpcom );
+
+	return Object.assign( wpcom, {
+		changeUser: function( username, password, fn ) {
+			return wpcom.req.post(
+				{
+					apiVersion: '1.1',
+					path: '/internal/support/' + username + '/grant'
+				},
+				{
+					password: password
+				},
+				( error, response ) => {
+					if ( ! error ) {
+						supportUser = response.username;
+						supportToken = response.token;
```

Comment:
In the future you may want to consider refactoring this so it doesn't need to depend on the closure, for easier testing, and for persistence through page reloads. I think this is fine for a first pass though.